src/Packages/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/build/Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props

```diff
@@ -62,21 +62,27 @@
                         using(var results = searcher.Get())
                           {
                             var mo = results.Cast<ManagementObject>().FirstOrDefault();
-                            Log.LogMessage("ExecutablePath is {0}", (string)mo["ExecutablePath"]);
-                            if(mo != null && ((string)mo["ExecutablePath"]).StartsWith(ImagePath, StringComparison.OrdinalIgnoreCase))
+                            if(mo != null)
                             {
-                              p.Kill();
-                              p.WaitForExit();
-                              Log.LogMessage("{0} is killed", (string)mo["ExecutablePath"]);
-                              break;
+                              var path = (string)mo["ExecutablePath"];
+                              var executablePath = path != null ? path : string.Empty;
+                              Log.LogMessage("ExecutablePath is {0}", executablePath);
+
+                              if(executablePath.StartsWith(ImagePath, StringComparison.OrdinalIgnoreCase))
+                              {
+                                p.Kill();
+                                p.WaitForExit();
+                                Log.LogMessage("{0} is killed", executablePath);
+                                break;
+                              }
                             }
                           }
                       }
                   }
                 }
                 catch (Exception ex)
                 {
-                  Log.LogErrorFromException(ex);
+                  Log.LogWarning(ex.Message);
```

Comment:
Why switching this to warning from error?  Isn't this suppose to be exception message?

Reply:
Error will cause build failure, while warning not necessarily will. The code here is to reduce the chance of copying Roslyn bits failure. If there is any exception, just log a warning which may not cause build failure.